### ISSUE TITLE
/moz/addon/install, invalid argument

### ISSUE THREAD

**Issue by wendao123** [mozilla/geckodriver#1436]
Hi, 
I could not install extension to firefox, is there something not set or wrong?

## System

* Version: 0.23
* Platform: Windows
* Firefox: 63.0.3 (64-bit)
* Selenium: 3.12

## Testcase
1. start a remote hub: 
    set webdriver.gecko.driver=geckodriver.exe
    START /B java -jar selenium-server-standalone-3.14.0.jar -role hub -port 4444 -timeout 3000
    START /B java -jar selenium-server-standalone-3.14.0.jar -role node -hub http://10.182.12.241:4444/grid/register -maxSession 50 -browser "browserName=firefox,maxInstances=50" -port 7777

2. connect to remote hub
webDriver = new RemoteWebDriver(new URL("http://10.182.12.241:4444/wd/hub"), DesiredCapabilities.firefox());

3. Install unsigned extension，

  JSONObject json = new JSONObject();
   json.put("addon", "C:/Users/wsc/Downloads/live-experience-js-demo-firefox.zip");
   json.put("temporary", true);
  CloseableHttpClient client = HttpClientBuilder.create().build();
  HttpPost httpPost = new HttpPost(“http://10.182.12.241:4444/wd/hub/session/" + 
  webDriver.getSessionId()+ "/moz/addon/install");
  httpPost.setEntity(new StringEntity(json.toString()));
  CloseableHttpResponse response = client.execute(httpPost);

## result
400 Bad Request
{
    "value": {
        "error": "invalid argument",
        "message": "Invalid byte 58, offset 1.",
        "stacktrace": ""
    }
}

**Comment by andreastt** [comment]
Could you please attach a [trace-level log] from geckodriver? Read more about reporting actionable bugs in our [contribution guidelines].

[trace-level log]: https://firefox-source-docs.mozilla.org/testing/geckodriver/geckodriver/TraceLogs.html
[contribution guidelines]: https://firefox-source-docs.mozilla.org/testing/geckodriver/geckodriver/Bugs.html

**Comment by wendao123** [comment]
i try to add below two lines to my client side code, but i do not know how to involve to DesiredCapabilities.firefox().
  FirefoxOptions options = new FirefoxOptions();
  options.setLogLevel(FirefoxDriverLogLevel.TRACE);
And as my environment is using hub, i also have no idea how to use options like "-vv"

**Comment by andreastt** [comment]
Yes, so roughly:

```java
JSONObject json = new JSONObject();
json.put("path", "C:/Users/wsc/Downloads/live-experience-js-demo-firefox.zip");
json.put("temporary", true);
```

**Comment by wendao123** [comment]
Sorry i was away and not reply on time.
In fact i tried "path" before i tried "addon", but "path" got another error, 

1543197264557	Marionette	TRACE	0 <- [1,1,null,{"sessionId":"924f871f-cc8c-463f-8b3c-eda98edf98f7","capabilities":{"browserName":"firefox","browserVersion":"63.0. ... AppData\\Local\\Temp\\rust_mozprofile.fBuu77k45dtK","moz:useNonSpecCompliantPointerOrigin":false,"moz:webdriverClick":true}}]
1543197264600	webdriver::server	DEBUG	<- 200 OK {"value":{"sessionId":"924f871f-cc8c-463f-8b3c-eda98edf98f7","capabilities":{"acceptInsecureCerts":true,"browserName":"firefox","browserVersion":"63.0.3","moz:accessibilityChecks":false,"moz:geckodriverVersion":"0.23.0","moz:headless":false,"moz:processID":737232,"moz:profile":"C:\\Users\\wsc\\AppData\\Local\\Temp\\rust_mozprofile.fBuu77k45dtK","moz:useNonSpecCompliantPointerOrigin":false,"moz:webdriverClick":true,"pageLoadStrategy":"normal","platformName":"windows","platformVersion":"6.1","rotatable":false,"setWindowRect":true,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"unhandledPromptBehavior":"dismiss and notify"}}}
09:54:24.600 INFO [ProtocolHandshake.createSession] - Detected dialect: W3C
09:54:24.647 INFO [RemoteSession$Factory.lambda$performHandshake$0] - Started new session 924f871f-cc8c-463f-8b3c-eda98edf98f7 (org.openqa.selenium.firefox.GeckoDriverService)
1543197264819	webdriver::server	DEBUG	-> POST /session/924f871f-cc8c-463f-8b3c-eda98edf98f7/window/maximize {}
1543197264803	Marionette	TRACE	0 -> [0,2,"WebDriver:MaximizeWindow",{}]
1543197264829	Marionette	TRACE	0 <- [1,2,null,{"x":-8,"y":-8,"width":1616,"height":876,"state":"maximized"}]
1543197264866	webdriver::server	DEBUG	<- 200 OK {"value":{"x":-8,"y":-8,"width":1616,"height":876}}
1543197264913	webdriver::server	DEBUG	-> POST /session/924f871f-cc8c-463f-8b3c-eda98edf98f7/url {"url":"about:blank"}
1543197264899	Marionette	TRACE	0 -> [0,3,"WebDriver:Navigate",{"url":"about:blank"}]
1543197264916	Marionette	DEBUG	[6442450945] Received DOM event beforeunload for about:blank
1543197264920	Marionette	DEBUG	[6442450945] Received DOM event pagehide for about:blank
1543197264928	Marionette	DEBUG	[6442450945] Received DOM event DOMContentLoaded for about:blank
1543197264938	Marionette	DEBUG	[6442450945] Received DOM event pageshow for about:blank
1543197264933	Marionette	TRACE	0 <- [1,3,null,{"value":null}]
1543197264944	webdriver::server	DEBUG	<- 200 OK {"value":null}
1543197276038	webdriver::server	DEBUG	-> POST /session/924f871f-cc8c-463f-8b3c-eda98edf98f7/moz/addon/install {"path":"C:/Users/wsc/Downloads/live-experience-js-demo-firefox.zip","temporary":true}
1543197276021	Marionette	TRACE	0 -> [0,4,"Addon:Install",{"path":"C:/Users/wsc/Downloads/live-experience-js-demo-firefox.zip","temporary":true}]
1543197276022	Marionette	TRACE	0 <- [1,4,{"error":"unknown error","message":"Expected absolute path: [Exception... \"Component returned failure code: 0x80520001  ... et@chrome://marionette/content/server.js:245:8\n_onJSONObjectReady/<@chrome://marionette/content/transport.js:490:9\n"},null]
1543197276038	webdriver::server	DEBUG	<- 500 Internal Server Error {"value":{"error":"unknown error","message":"Expected absolute path: [Exception... \"Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIFile.initWithPath]\"  nsresult: \"0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)\"  location: \"JS frame :: chrome://marionette/content/addon.js :: install :: line 62\"  data: no]","stacktrace":"WebDriverError@chrome://marionette/content/error.js:178:5\nUnknownError@chrome://marionette/content/error.js:483:5\ninstall@chrome://marionette/content/addon.js:64:13\nGeckoDriver.prototype.installAddon@chrome://marionette/content/driver.js:3358:10\ndespatch@chrome://marionette/content/server.js:298:20\nexecute@chrome://marionette/content/server.js:271:11\nonPacket/<@chrome://marionette/content/server.js:246:15\nonPacket@chrome://marionette/content/server.js:245:8\n_onJSONObjectReady/<@chrome://marionette/content/transport.js:490:9\n"}}

**Comment by wendao123** [comment]
i know the reason, 
json.put("path", "C:/Users/wsc/Downloads/live-experience-js-demo-firefox.zip") need to replace with
json.put("path", "C:\\Users\\wsc\\Downloads\\live-experience-js-demo-firefox.zip");

**Comment by andreastt** [comment]
Yes, whereas forward slashes may be recognised in Java, they are very likely not by geckodriver which is written in Rust.

Thanks for confirming that you were able to fix your issue!
